libraries/AP_Compass/AP_Compass_AK8963.cpp

```diff
@@ -155,17 +154,18 @@ bool AP_Compass_AK8963::init()
     set_dev_id(_compass_instance, _dev_id);
 
     /* timer needs to be called every 10ms so set the freq_div to 10 */
-    _timesliced = hal.scheduler->register_timer_process(FUNCTOR_BIND_MEMBER(&AP_Compass_AK8963::_update, void), 10);
+    if (!_bus->register_periodic_callback(10000, FUNCTOR_BIND_MEMBER(&AP_Compass_AK8963::_update, bool))) {
```

Comment:
this looks like the correct change, calling `register_periodic_callback` in BusDriver, not on Device. I wonder if the change in the previous commit in MPU6000 is just a leftover.